From f676b86ff0cdd694f40adac0170ab717881b81c9 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 3 Oct 2014 16:24:22 -0700
Subject: [PATCH 488/793] drm: xilinx: dp: Fix the clock clean up code

Call clk_disable_unprepare() on error exit in xilinx_drm_dp_probe,
and change the order of calling clean up functions.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_dp.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
index 7793407..b466800 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
@@ -1261,8 +1261,10 @@ static int xilinx_drm_dp_probe(struct platform_device *pdev)
 
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	dp->iomem = devm_ioremap_resource(dp->dev, res);
-	if (IS_ERR(dp->iomem))
-		return PTR_ERR(dp->iomem);
+	if (IS_ERR(dp->iomem)) {
+		ret = PTR_ERR(dp->iomem);
+		goto error_clk;
+	}
 
 	platform_set_drvdata(pdev, dp);
 
@@ -1325,6 +1327,8 @@ static int xilinx_drm_dp_probe(struct platform_device *pdev)
 
 error:
 	drm_dp_aux_unregister(&dp->aux);
+error_clk:
+	clk_disable_unprepare(dp->aclk);
 	return ret;
 }
 
@@ -1332,12 +1336,12 @@ static int xilinx_drm_dp_remove(struct platform_device *pdev)
 {
 	struct xilinx_drm_dp *dp = platform_get_drvdata(pdev);
 
-	clk_disable_unprepare(dp->aclk);
-
 	xilinx_drm_writel(dp->iomem, XILINX_DP_TX_ENABLE, 0);
 
 	drm_dp_aux_unregister(&dp->aux);
 
+	clk_disable_unprepare(dp->aclk);
+
 	return 0;
 }
 
-- 
1.7.5.4

